curricula/templates/curricula/curriculum.html

```diff
@@ -217,6 +221,33 @@ <h2>{{ topic.name }}</h2>
                 });
                 */
             });
+            $("#clone_this").click(function () {
+                $('#clone_this').html("Cloning...").removeClass('btn-warning btn-success').addClass('btn-primary');
+                var pk = $(this).attr('data-pk');
+                var type = $(this).attr('data-type');
+                var children = $('#clone_children').prop('checked');
+                $('#clone_spinner').addClass('fa fa-cog fa-spin');
+
+                $.ajax({
+                     type: "POST",
+                     url: "/clone/",
+                     data: {pk: pk, type: type, children: children},
+                     timeout: 9999999
+                }).done(function (response) {
+                    $("#clone_this").addClass('btn-success').text("Success");
+                    $("#clone_results").text(JSON.stringify(response, undefined, 2));
+                    $('#clone_spinner').removeClass('fa fa-cog fa-spin');
+                    if (response.hasOwnProperty('redirect_url')) {
+                        setTimeout(function () {
+                            window.location.href = response.redirect_url;
+                        }, 500);
+                    }
+                }).fail(function (response) {
+                    $("#clone_this").addClass('btn-warning').text("Failed");
+                    $("#clone_results").text(JSON.stringify(response, undefined, 2));
+                    $('#clone_spinner').removeClass('fa fa-cog fa-spin');
+                });
+            });
```

Comment:
looks like this chunk of code gets repeated quite a bit throughout the templates; can it be extracted into a partial?

Reply:
Yup, I've got a todo in unit.html to clean that up. There's a bunch of refactoring I want to do in the templates when I have some time.